pkg/orchestrator/endpoint.go

```diff
@@ -177,6 +252,82 @@ func (e *BaseEndpoint) StopJob(ctx context.Context, request *StopJobRequest) (St
 	}, nil
 }
 
+func (e *BaseEndpoint) RerunJob(ctx context.Context, request *RerunJobRequest) (*RerunJobResponse, error) {
+	txContext, err := e.store.BeginTx(ctx)
+	job, err := e.store.GetJobByIDOrName(txContext, request.JobIDOrName, request.Namespace)
+	if err != nil {
+		return nil, jobstore.NewJobStoreError(err.Error())
+	}
+	if request.JobVersion != 0 {
+		job, err = e.store.GetJobVersion(txContext, job.ID, request.JobVersion)
+		if err != nil {
+			return nil, jobstore.NewJobStoreError(err.Error())
+		}
+	}
+
+	defer txContext.Rollback() //nolint:errcheck
+
+	switch job.State.StateType {
+	case models.JobStateTypePending, models.JobStateTypeQueued, models.JobStateTypeUndefined:
+		return nil, bacerrors.Newf("cannot rerun job in state %s", job.State.StateType)
+	default:
+		// continue
+	}
+
+	if err = e.store.UpdateJob(txContext, job); err != nil {
+		return nil, err
+	}
+
+	if err = e.store.UpdateJobState(txContext, jobstore.UpdateJobStateRequest{
```

Comment:
If we move this to update job, no need to add it here anymore, same as the submit job

Reply:
Added to issue `ENG-855`